app.js

```diff
@@ -32,10 +33,10 @@ const adminAppRouter = express.Router();
  * @summary Middlewares
  */
 if (logger) app.use(logger);
-app.use(basePath, appRouter);
+app.use(`/api${basePath}`, appRouter);
 appRouter.use(authentication);
 
-adminApp.use(basePath, adminAppRouter);
+adminApp.use(`/api${basePath}`, adminAppRouter);
```

Comment:
Have a question about this change. Instead of reading`basePath` from the configuration file, the API is now getting this field from the swagger file. However, I'm not sure what value should I put in this field. Based on the [swagger definition](https://swagger.io/docs/specification/2-0/api-host-and-base-path/), the `basePath` should only be like `/v1`.  SmartDoc will also construct the API URL automatically by concatenating `schemes`, `host` and `basePath`. However, in our current DW swagger, the `basePath` has a leading `/api` like [here](https://github.com/osu-mist/web-api-skeleton/blob/master/swagger.yaml#L15). It will make SmartDoc create an API URL be like `https://api.oregonstate.edu/api/v1` which is not correct. I'm not sure will that be safe to remove this `/api` from the `basePath`? Or is there any reason we should keep this prefix? cc @jaredkosanovic

Reply:
Good question. This brings up another question I have, which is how do we differentiate API specs between what the client should expect, and what our server should provide? For example, a few of our APIs expect the `OSU-API-act-as` header to know which user is making the API call. Only apigee should use that header. The clients should not be aware of it. It almost seems like we need two separate API specs - one for the client and one for the server, but that doesn't seem right.